pandas/core/groupby/groupby.py

```diff
@@ -343,14 +345,25 @@ def _group_selection_context(groupby):
     groupby._reset_group_selection()
 
 
+KeysArgType = Optional[
```

Comment:
in general maybe better not to include the Optional in the alias, and just add the Optional where needed when using the alias. I find this promotes better resuse of aliases.

Reply:
Yea I went back and forth on this myself. What swayed me to include it here is that outside of the signature it is valid for this object to always be `None`, so long as it is paired with a valid `level`. But happy to change

Reply:
also I _think_ aliases should always start with an underscore.